### PR TITLE
Feature/deploy new contracts 20230902

### PR DESCRIPTION
**Pull request by zodahu**
Router: 0x4E744c3E6973D34ee130B7E668Abc14CD49ca16e
AaveV2FlashLoanCallback: 0xD1CA91bE788372275FB0FfC876465Bc0a5A31F86
AaveV3FlashLoanCallback: 0x8f2Ed9cE5DF73210c5Fa21d0cFDFF98bB1027a1F
BalancerV2FlashLoanCallback: 0x13431cd779FD770D55701B96F2675dFF63BDD756
Create3Factory: 0xB9504E656866cCB985Aa3f1Af7b8B886f8485Df6

### CODE REVIEW

**Review by hihiben — APPROVED** [review]
LGTM

File: script/DeployRouter.s.sol [hunk 1]
```diff
@@ -8,14 +8,18 @@ import {Router} from 'src/Router.sol';
 import {DeployBase} from './DeployBase.s.sol';
 
 abstract contract DeployRouter is DeployBase {
+    address public constant SIGNER = 0xffFf5a88840FF1f168E163ACD771DFb292164cFA;
```
Comment (Jeff-CCH) [inline script/DeployRouter.s.sol:11]: Would we always use the same signer for multi-chains? If the signer would change, is it better to pass it as a param?

File: script/DeployRouter.s.sol [hunk 2]
```diff
@@ -27,7 +31,7 @@ abstract contract DeployRouter is DeployBase {
         deployedAddress = cfg.deployedAddress;
         if (deployedAddress == UNDEPLOYED) {
             ICREATE3Factory factory = ICREATE3Factory(create3Factory);
-            bytes32 salt = keccak256('protocolink.router.v2');
+            bytes32 salt = keccak256('protocolink.router.v1');
```
Comment (Jeff-CCH) [inline script/DeployRouter.s.sol:34]: I thought the version number was an increment. Is there any reason why we decrease the version number for this deployment?

File: script/DeployRouter.s.sol [hunk 3]
```diff
@@ -36,6 +40,16 @@ abstract contract DeployRouter is DeployBase {
...
             console2.log('Router Owner:', router.owner());
+
+            // Set and check signer
+            router.addSigner(SIGNER);
+            require(router.signers(SIGNER), 'Router signer is invalid');
+
+            // Set and check fee rate
+            if (cfg.feeRate > 0) {
+                router.setFeeRate(cfg.feeRate);
+                require(router.feeRate() == cfg.feeRate, 'Router fee rate is invalid');
```
Comment (Jeff-CCH) [inline script/DeployRouter.s.sol:51]: wondering if this `require` should be placed after the `if(cfg.feeRate > 0)` clause to ensure `router.feeRate() == 0`

File: script/DeployRouter.s.sol (hunk 1 above)
Comment (zodahu) [reply script/DeployRouter.s.sol]: Fixed, thanks.

File: script/DeployRouter.s.sol (hunk 3 above)
Comment (zodahu) [reply script/DeployRouter.s.sol]: Fixed, thanks.

File: script/DeployRouter.s.sol (hunk 2 above)
Comment (zodahu) [reply script/DeployRouter.s.sol:33]: As the create3 factory was redeployed using a new testing deployer, we can reset the salt.

**Review by Jeff-CCH — APPROVED** [review]
LGTM

**Pull request merged**